src/instruction/if_else.rs

```diff
@@ -81,6 +83,15 @@ impl Instruction for IfElse {
 
 impl TypeCheck for IfElse {
     fn resolve_type(&self, ctx: &mut TypeCtx) -> CheckedType {
+        let bool_checkedtype = CheckedType::Resolved(TypeId::from("bool"));
+        let cond_ty = self.condition.resolve_type(ctx);
+        if cond_ty != bool_checkedtype {
+            ctx.error(Error::new(ErrKind::TypeChecker).with_msg(format!(
+                "if condition should be a boolean, not a {}",
```

Comment:
```suggestion
                "if condition should be a boolean, not a `{}`",
```

Tiny nit :P this is what we do in the rest of the type checking errors for now

Reply:
Fixed by 16a03e4